ui/src/main.rs

```diff
@@ -966,6 +974,70 @@ fn get_update_icon(cx: Scope) -> Element {
     }
 }
 
+#[inline_props]
+pub fn get_download_modal<'a>(
+    cx: Scope<'a>,
+    on_submit: EventHandler<'a, PathBuf>,
+    on_dismiss: EventHandler<'a, ()>,
+) -> Element<'a> {
+    let download_location: &UseState<Option<PathBuf>> = use_state(cx, || None);
+
+    let dl = download_location.current();
+    let disp_download_location = dl
+        .as_ref()
+        .clone()
+        .map(|x| x.to_string_lossy().to_string())
+        .unwrap_or_default();
+
+    cx.render(rsx!(Modal {
+        on_dismiss: move |_| on_dismiss.call(()),
+        children: cx.render(rsx!(
+            div {
+            class: "download-modal flex col",
+            h1 {
+                get_local_text("updates.title")
+            },
+            ul {
+                class: "instruction-list",
+                li {
+                    get_local_text("updates.instruction1")
+                },
+                li {
+                    Button {
+                        text: get_local_text("updates.download-label"),
+                        aria_label: get_local_text("updates.download-label"),
+                        appearance: Appearance::Secondary,
+                        onpress: |_| {
+                            let _ = open::that("https://github.com/Satellite-im/Uplink/releases/latest");
+                        }
+                    }
+                },
+                li {
+                    get_local_text("updates.instruction2")
+                },
+                li {
+                    get_local_text("updates.instruction3")
+                },
+                li {
+                    get_local_text("updates.instruction4")
+                }
+            },
+            p {
+                "*We are going to streamline this process in a future update."
+            },
+            // dl.as_ref().clone().map(|dest| rsx!(
+            //     Button {
+            //         text: "download installer".into(),
+            //         onpress: move |_| {
+            //            on_submit.call(dest.clone());
+            //         }
+            //     }
+            // ))
```

Comment:
do we need this commented out code? 🔨

Reply:
we do, I didnt want to rip it out, it just wasn't working right now and I didnt have time to look into why